apstools/devices/motor_factory.py

```diff
@@ -0,0 +1,136 @@
+"""
+Motor Bundle Factory
+====================
+
+Factory for creating bundles with any number of positioners.
+
+.. autosummary::
+
+    ~axis_component
+    ~mb_class_factory
+    ~mb_creator
+
+.. rubric:: Examples
+
+.. code-block:: python
+    :linenos:
+
+    xy_stage = mb_creator(prefix="IOC:", motors={"x": "m21", "y": "m22"})
+
+.. seealso:: :doc:`How to use 'mb_creator()' </examples/ho_motor_factory>`
+"""
+
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+from typing import Type
+from typing import Union
+
+from ophyd import Component
+from ophyd import EpicsMotor
+from ophyd import Kind
+from ophyd import MotorBundle
+from ophyd import SoftPositioner
+
+from apstools.utils import dynamic_import
+
+DEFAULT_DEVICE_BASE_CLASSES: List[Union[str, Type]] = [MotorBundle]
+DEFAULT_DEVICE_CLASS_NAME: str = "MB_Device"
+DEFAULT_MOTOR_LABELS: List[str] = ["motors"]
+DEFAULT_KIND: Kind = Kind.hinted | Kind.normal
+
+MOTORS_TYPE = Union[List[str], Dict[str, Optional[Union[str, Dict]]]]
+
+
+def axis_component(parms: dict[str, Any]) -> Component:
+    """Return a Component with 'parms for this axis."""
+    parms = parms or {}
+    if isinstance(parms, str):
```

Comment:
This doesn't match the type hint. If parms can be a string, then the type hint should be ``str | dict[str, Any]``.

Reply:
Perha[s even `Union[None, str, Dict[str, Any]]`?

Reply:
My solution would be `Union[str, Mapping[str, Any]] = {}` and get rid of line 48, but there may be reasons to disfavor the mutable default argument.

Reply:
I expanded the typing but left the `None` arg, based on static code review.